WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Service/BlogDashboardPostsParser.swift

```diff
@@ -19,19 +19,25 @@ class BlogDashboardPostsParser {
             return postsDictionary
         }
 
-        if !posts.hasDrafts {
-            // Check for local drafts
-            if let localDraftsCount = numberOfPosts(for: blog, filter: PostListFilter.draftFilter()),
-               localDraftsCount > 0 {
-                posts["draft"] = [0...localDraftsCount].map { _ in [:] }
+        if let localDraftsCount = numberOfPosts(for: blog, filter: PostListFilter.draftFilter()) {
+            if blog.dashboardState.draftsSynced { // If drafts are synced, depend on local data
+                posts["draft"] = Array(repeatElement([:], count: localDraftsCount))
+            }
+            else { // If drafts are not synced, only depend on local data if the cards API returns zero posts
+                if !posts.hasDrafts, localDraftsCount > 0 {
+                    posts["draft"] = Array(repeatElement([:], count: localDraftsCount))
+                }
             }
```

Comment:
What do you think of getting rid of the else block and combining into one block?

```suggestion
            // If drafts are synced OR
            // If drafts are not synced and the cards API returns zero posts
            // depend on local data
            if blog.dashboardState.draftsSynced || (!posts.hasDrafts && localDraftsCount > 0) { 
                posts["draft"] = Array(repeatElement([:], count: localDraftsCount))
            }
```

Reply:
I split them to enhance readability. But I think that comment now clears everything up 👍 
Fixed in 652675f769f219409308953b9b46b03eec564d25